### ISSUE TITLE
JitAllocator.alloc arguments?

### ISSUE THREAD

**Issue by RazvanAciu** [asmjit/asmjit#239]
Hi, 
I tried to use JitAllocator.alloc, but I saw that it needs 3 arguments.
Can you update the example or explain how these are used?

Thank you,
Razvan

**Comment by kobalicek** [comment]
If you are talking about this function:

```c++
Error JitAllocator::alloc(void** roPtrOut, void** rwPtrOut, size_t size) noexcept;
```

it returns two pointers, one read only, which is executable, and one read/write, which can be used to store code into the executable memory. If you don't use `kOptionUseDualMapping` option then these two pointers would be the same. If you use `kOptionUseDualMapping` option and the operating system supports dual mapping then the pointers would be different, but would point into the same mapped mempry.

**Comment by RazvanAciu** [comment]
Thank you.
For now I used a workaround: I declared a global JitRuntime managed through shared_ptr's, so I can be sure it outlasts any object that is using it (I am using global objects without a deterministic destruction order). When I jit a function, I get a new shared_ptr reference to this JitRuntime. When I don't need the function anymore, I release it from the runtime and I also release its runtime shared_ptr.
